shared/subspace-data-retrieval/src/piece_fetcher.rs

```diff
@@ -28,7 +28,7 @@ use tracing::{debug, trace};
 // This code was copied and modified from subspace_service::sync_from_dsn::download_and_reconstruct_blocks():
 // <https://github.com/autonomys/subspace/blob/d71ca47e45e1b53cd2e472413caa23472a91cd74/crates/subspace-service/src/sync_from_dsn/import_blocks.rs#L236-L322>
 pub async fn download_pieces<PG>(
-    piece_indexes: Vec<PieceIndex>,
+    piece_indexes: &Vec<PieceIndex>,
```

Comment:
Should be `&[PieceIndex]` in such cases

Reply:
Yeah I tried that and something weird happened with type inference. Happy to have another go next time around.